pkg/controller/vpc_nat.go

```diff
@@ -17,11 +21,25 @@ func (c *Controller) resyncVpcNatImage() {
 		klog.Error(err)
 		return
 	}
+
 	image, exist := cm.Data["image"]
 	if !exist {
 		err = fmt.Errorf("%s should have image field", util.VpcNatConfig)
 		klog.Error(err)
 		return
 	}
 	vpcNatImage = image
+
+	// Check BGP is enabled on the NAT GW, if yes, verify required parameters are present
+	enableBgpSpeaker, exist := cm.Data["enableBgpSpeaker"]
+	if exist && enableBgpSpeaker == "true" {
+		vpcNatGwEnableBgpSpeaker = true
```

Comment:
I think we should move this switch into the vpc nat gateway CRD.

Reply:
it's mainly here as an ultimate failswitch because this is experimental, I guess we can get rid of it, the code seems safe enough to not accidentally inject the speaker sidecar

Reply:
I removed it entirely